### PR TITLE
Streamline network logging - drop scrape ID and only log on errors

### PR DESCRIPTION
**Pull request by tuommaki**
Scrape ID w/ debug level logs is spamming a lot and doesn't really provide much
valuable information. Generally `net-exporter` logs are only interesting when
there's a problem so it makes sense to log only in those cases. Scrape ID
doesn't really provide additional information in those cases but it's more
interesting what's the problem and with what peer.

Towards https://github.com/giantswarm/giantswarm/issues/9122

### CODE REVIEW

File: network/network.go [hunk 1]
```diff
@@ -300,7 +279,7 @@ func (c *Collector) getNeighbours(n int, subsets []v1.EndpointSubset) ([]string,
 	// Calculate n neighbours, given our local IP and all other net-exporter IPs.
 	neighbours := c.calculateNeighbours(n, ip, addresses)
 
-	c.logger.Log("level", "info", "message", "calculated neighbours", "ip", ip, "neighbours", strings.Join(neighbours, ", "), "scrapeID", c.scrapeID)
+	c.logger.Log("level", "info", "message", "calculated neighbours", "ip", ip, "neighbours", strings.Join(neighbours, ", "))
```
Comment (tuommaki) [inline network/network.go]: Unsure if this should be dropped as well?

Comment (JosephSalisbury) [reply network/network.go]: I'd drop it, I feel like I added this for debug

Comment (tuommaki) [reply network/network.go]: I'll leave it for now. It's still a bit spammy, but I have a strong feel that this information will be useful one day when diagnosing problems. We can drop it if it's annoying.

**Comment by kopiczko** [comment]
> Scrape ID w/ debug level logs is spamming a lot and doesn't really provide much
> valuable information. Generally `net-exporter` logs are only interesting when
> there's a problem so it makes sense to log only in those cases. Scrape ID
> doesn't really provide additional information in those cases but it's more
> interesting what's the problem and with what peer.

We added that back then with Ferran because there are many go routines flying around. If the timeouts are not handled properly logs from different scrape request can intertwine. Then it makes it difficult to debug.

**Pull request merged**